.travis.yml

```diff
@@ -1,5 +1,21 @@
 language: haskell
 
+ghc: 7.8
+
+addons:
+  apt:
+    packages:
+    - ocaml
+    - camlidl
+
+sudo: false
+
+cache:
+  apt: true
+  directories:
```

Comment:
I've been reluctant to enable caching for cabal packages because then we have to worry about dependency conflicts as time moves on.

At the very least we need to add `--upgrade-dependencies` to the `cabal install` invocations to ensure we're always testing against new versions of packages. Is there another flag we can pass to ensure a more sandbox-like environment?

(I've been considering setting up the nix package manager as a first step on travis to improve build times, since it has binary caches for haskell packages.)